### PR TITLE
button toggle states

### PR DESCRIPTION
**Pull request by jhardy**
### :sparkles: Changes

This PR updates button toggles and button groups to have a hover state that is not based on the key color. This helps differentiate selected vs unselected items. It also adds focus visible functionality to the button items

#### Before
image

#### After
image

### :white_check_mark: Requirements

- [ ] Includes test coverage for all changes
- [ ] Build and tests are passing
- [ ] Update documentation
- [ ] Updated CHANGELOG
- [ ] Checked for i18n impacts
- [ ] Checked for a11y impacts
- [ ] Check for image-snapshot changes (run `yarn image-snapshots` locally)
- [ ] PR is ideally < ~400LOC

### CODE REVIEW

File: packages/components/src/Button/ButtonItem.tsx [hunk 1]
```diff
@@ -39,17 +39,30 @@ export interface ButtonItemProps
   extends SpaceProps,
     Omit<CompatibleHTMLProps<HTMLButtonElement>, 'type' | 'aria-pressed'> {
   value?: string
+  focusVisible?: boolean
```
Comment (jhardy) [inline packages/components/src/Button/ButtonItem.tsx:42]: Add focus visible behavior

File: packages/components/src/Button/ButtonItem.tsx [hunk 2]
```diff
@@ -67,23 +80,31 @@ const ButtonLayout = forwardRef(
...
 )
 
 ButtonLayout.displayName = 'ButtonLayout'
 
+const ButtonOuter = styled.button<ButtonItemProps>`
+  ${(props) =>
+    props.focusVisible &&
+    `box-shadow: 0 0 0.5px 1px ${props.theme.colors.keyFocus}`}
+`
+
```
Comment (jhardy) [inline packages/components/src/Button/ButtonItem.tsx:107]: new component here so apply focus visible box shadow

File: packages/components/src/Button/ButtonItem.tsx [hunk 3]
```diff
@@ -105,11 +126,11 @@ export const ButtonItem = styled(ButtonLayout)`
   &:active,
   &:focus,
   &:hover {
-    background: ${({ theme }) => theme.colors.keyAccent};
+    background: ${({ theme }) => theme.colors.neutralSubtle};
+    color: ${({ theme }) => theme.colors.text5};
```
Comment (jhardy) [inline packages/components/src/Button/ButtonItem.tsx:130]: Update to neutral subtle and text on hover

**Review by lukelooker — APPROVED** [review]
LGTM

**Pull request merged**